worker/cppworker/worker/OCCChild.cpp

```diff
@@ -1524,6 +1524,39 @@ int OCCChild::connect(const std::string& db_username, const std::string& db_pass
 	//create the user session
 	rc = OCISessionBegin(svchp,	 errhp, authp, OCI_CRED_RDBMS, 
 			(ub4) OCI_DEFAULT);
+	if(rc!=OCI_SUCCESS) {
+		std::string db_pwd2 = getenv("password2");
```

Comment:
What if we used 

envStr[12]="password";
for (int i=0; i<RETRIES; i++) {
    if (i > 0) sprintf(envStr, "password%d", i);
    std::string db_pwd = getenv(envStr);
    // OCIAttrSet password
    // OCISessionBegin
    if (rv==OCI_SUCCESS) {
        break;

I think we'll keep the code readable and consolidate some code.

Reply:
Sure, Let me update it